gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -175,18 +176,20 @@ public void startUpload(PipedInputStream pipeSource) {
     }
 
     @Override
-    public Object call() throws IOException, InterruptedException {
+    public Object call() throws IOException {
       // Try-with-resource will close this end of the pipe so that
       // the writer at the other end will not hang indefinitely.
       try (InputStream toClose = pipeSource) {
         return doResumableUpload();
+      } catch (IOException e) {
+        throw e;
       } catch (Exception e) {
```

Comment:
Are there any non-IOExceptions that thrown in `doResumableUpload` method by gRPC?

Reply:
Not to my knowledge. I can remove the clause.